modules/rubiconAnalyticsAdapter.js

```diff
@@ -51,27 +51,16 @@ const cache = {
 
 const BID_REJECTED_IPF = 'rejected-ipf';
 
-let fpkvs = {};
-function updateFpkvs(fpkvs, newKvs) {
-  const isValid = typeof newKvs === 'object' && Object.keys(newKvs).every(key => typeof newKvs[key] === 'string');
-  if (!isValid) {
-    utils.logError('Rubicon Analytics: fpkvs must be object with string keys and values');
-    return fpkvs;
-  } else {
-    return {...fpkvs, ...newKvs};
-  }
-}
-
-let integration, ruleId, wrapperName;
-// listen for any rubicon setConfig events and save them to appropriate fields!
+export let rubiConf = {
+  pvid: utils.generateUUID().slice(0, 8)
+};
 // we are saving these as global to this module so that if a pub accidentally overwrites the entire
 // rubicon object, then we do not lose other data
 config.getConfig('rubicon', config => {
-  let rubiConf = config.rubicon;
-  integration = rubiConf.int_type || integration || DEFAULT_INTEGRATION;
-  ruleId = rubiConf.rule_name || ruleId;
-  wrapperName = rubiConf.wrapperName || wrapperName;
-  fpkvs = rubiConf.fpkvs ? updateFpkvs(fpkvs, rubiConf.fpkvs) : fpkvs
+  utils.mergeDeep(rubiConf, config.rubicon);
+  if (utils.deepAccess(config, 'rubicon.updatePageView') === true) {
+    rubiConf.pvid = utils.generateUUID().slice(0, 8)
+  }
```

Comment:
deep merge everything, but if this specific setConfig call wants us to regenerate the pvid, we do so.